websites/A/Anime Genzai/presence.ts

```diff
@@ -0,0 +1,226 @@
+const presence = new Presence({
+		clientId: "1071365131616915517",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+async function getStrings() {
+	return presence.getStrings(
+		{
+			anime: "general.anime",
+			viewHome: "general.viewHome",
+			viewing: "general.viewing",
+			search: "general.search",
+			searchFor: "general.searchFor",
+			play: "general.watchingVid",
+			pause: "general.paused",
+			searchSomething: "general.searchSomething",
+			watchingMovie: "general.watchingMovie",
+			watchVideoButton: "general.buttonWatchVideo",
+			buttonViewPage: "general.buttonViewPage",
+			viewPage: "general.viewPage",
+			viewMovie: "general.viewMovie",
+			chapter: "general.chapter",
+			viewThread: "general.viewThread",
+			viewProfile: "general.viewProfile",
+		},
+		await presence.getSetting<string>("lang").catch(() => "es")
+	);
+}
+
+enum Assets {
+	Logo = "https://i.imgur.com/dtmYEl8.png",
+	Reading = "https://i.imgur.com/wPUmqu5.png",
+	Searching = "https://i.imgur.com/UhPgTRn.png",
+	Viewing = "https://i.imgur.com/RMg2Qgg.png",
+	Play = "https://i.imgur.com/juDzimk.png",
+	Pause = "https://i.imgur.com/GbAMPHI.png",
+}
+
+let video = {
+		duration: 0,
+		currentTime: 0,
+		paused: true,
+	},
+	strings: Awaited<ReturnType<typeof getStrings>>;
+
+function textContent(tags: string) {
+	return document.querySelector(tags)?.textContent?.trim();
+}
+
+function getImage(tags: string) {
+	return document
+		.querySelector<HTMLImageElement>(tags)
+		?.style?.backgroundImage?.slice(4, -1)
+		?.replace(/"/g, "");
+}
+
+presence.on(
+	"iFrameData",
+	(data: { duration: number; currentTime: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+		},
+		[privacy, logo, time, buttons] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("logo"),
+			presence.getSetting<boolean>("time"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		{ pathname, href } = document.location,
+		path = pathname.split("/")[1];
+
+	if (!strings) strings = await getStrings();
```

Comment:
This is a better way to do this (3/3):
```suggestion
	if (oldLang !== newLang || !strings) {
		oldLang = newLang;
		strings = await getStrings();
	}
```